lib/internal.h

```diff
@@ -347,32 +347,19 @@ static inline int io_getevents(__attribute__((unused)) aio_context_t ctx,
  * Auxiliary macros
  ************************************************************/
 
-#if __GNUC__ >= 4
-# define DSO_PUBLIC __attribute__ ((visibility ("default")))
-
 #if __GNUC__ >= 10
 # define IMPL_SYMVER(name, version) \
-    __attribute__ ((visibility ("default"))) \
     __attribute__((__symver__("kcapi_" #name "@@LIBKCAPI_" version)))
 
 # define ORIG_SYMVER(name, version) \
-    __attribute__ ((visibility ("default"))) \
     __attribute__((__symver__("kcapi_" #name "@LIBKCAPI_" version)))
 
 #else
 # define IMPL_SYMVER(name, version) \
-    __asm__(".global impl_" #name ";"\
-	    ".symver impl_" #name ",kcapi_" #name "@@LIBKCAPI_" version);\
-    __attribute__ ((visibility ("default")))
+    __asm__(".symver impl_" #name ",kcapi_" #name "@@LIBKCAPI_" version);
 
 # define ORIG_SYMVER(name, version) \
-    __asm__(".global orig_" #name ";"\
-	    ".symver orig_" #name ",kcapi_" #name "@LIBKCAPI_" version);\
-    __attribute__ ((visibility ("default")))
-#endif
-
-#else
-# error "Compiler version too old"
```

Comment:
May I as why you dropped the #error statement here?

Reply:
Didn't think it was necessary as we are not using the visibility attribute anymore.
That's what the previous code was guarding about.